### DISCUSSION TITLE
Context conflict when handling multiple Nuxt apps via `nuxt/kit`

### DISCUSSION THREAD

**Discussion by Jinjiang** [nuxt/nuxt#27608] [Questions] [answered]
Reproduction:
https://github.com/Jinjiang/reproductions/tree/nuxtjs-conflict-20240614

I was trying to `nuxt/kit` to prepare 3 Nuxt apps in one process. Then I failed and found the context conflict error.

Question:

1. Is the way I use `nuxt/kit` correst? At least it works if I only prepare one Nuxt app.
2. Is there anyway we can make it? And I guess it's also the same issue if I run the tests or do other stuff?

Thanks.

**Comment by danielroe** [comment, marked as answer]
They will need to be loaded sequentially, or in separate JS contexts (child processes or worker threads, etc.). The Nuxt builder assumes it is a singleton in programmatic usage, which enables the usage of `useNuxt` in modules, etc.
